l3_dump.py

```diff
@@ -8,18 +8,67 @@
 import sys
 import struct
 import subprocess
+import os
 
 if len(sys.argv) != 3:
     print(f"Usage: {sys.argv[0]} <logfile> <program-binary>")
+    print(" ")
+    print("Note: If L3 logging is done for <program-binary> with L3_LOC_ENABLED=1")
+    print("we expect to find the LOC-decoder binary named <program-binary>_loc")
     sys.exit(0)
 
+# #############################################################################
+# Check for required LOC-decoder binary, based on environment of run
+# If L3_LOC_ENABLED=1 is set in the env, search for a LOC-decoder binary
+# named "<program_binary>_loc".
+# #############################################################################
+PROGRAM_BIN = sys.argv[2]
+DECODE_LOC_ID = 0   # By default, we expect L3 logging was done w/LOC OFF.
+LOC_ENABLED = "L3_LOC_ENABLED"
+LOC_DECODER = "LOC_DECODER"
+if LOC_ENABLED in os.environ:
+    env_var_val = os.getenv(LOC_ENABLED)
+    if env_var_val == "1":
+        LOC_DECODER = PROGRAM_BIN + "_loc"
+        if os.path.exists(LOC_DECODER) is False:
+            print(f"Env-var {LOC_ENABLED}=1 is set, but required "
+                  + f"LOC-decoder binary {LOC_DECODER} is not found.")
+            sys.exit(1)
+
+        # LOC is in-use and LOC-decoder binary was found.
+        DECODE_LOC_ID = 1
+
+# #############################################################################
+def exec_binary(cmdargs:list):
+    """Execute a binary, with args, and print output to stdout."""
+    try:
+        with subprocess.Popen(cmdargs, stdout=subprocess.PIPE, text=True) as output:
+            sp_stdout, sp_stderr = output.communicate()
```

Comment:
Please review this carefully. I cloned this what's been done elsewhere, and from other Python scripts written by me in the past.

I think this works ... tests fine. Kindly review.

Reply:
Looks ok to me!